src/m365/spo/commands/applicationcustomizer/applicationcustomizer-set.ts

```diff
@@ -0,0 +1,185 @@
+      {
+        option: '-i, --id [id]'
+      },
+      {
+        option: '-c, --clientSideComponentId [clientSideComponentId]'
+      },
+      {
+        option: '--newTitle [newTitle]'
+      },
+      {
+        option: '--clientSideComponentProperties [clientSideComponentProperties]'
+      },
+      {
+        option: '-s, --scope [scope]', autocomplete: this.allowedScopes
+      }
+    );
+  }
+
+  #initTelemetry(): void {
+    this.telemetry.push((args: CommandArgs) => {
+      Object.assign(this.telemetryProperties, {
+        title: typeof args.options.title !== 'undefined',
+        id: typeof args.options.id !== 'undefined',
+        clientSideComponentId: typeof args.options.clientSideComponentId !== 'undefined',
+        newTitle: typeof args.options.newTitle !== 'undefined',
+        clientSideComponentProperties: typeof args.options.clientSideComponentProperties !== 'undefined',
+        scope: typeof args.options.scope !== 'undefined'
+      });
+    });
+  }
+
+  #initValidators(): void {
+    this.validators.push(
+      async (args: CommandArgs) => {
+        if (args.options.id && !validation.isValidGuid(args.options.id)) {
+          return `${args.options.id} is not a valid GUID`;
+        }
+
+        if (args.options.clientSideComponentId && !validation.isValidGuid(args.options.clientSideComponentId)) {
+          return `${args.options.clientSideComponentId} is not a valid GUID`;
+        }
+
+        if (args.options.scope && this.allowedScopes.indexOf(args.options.scope) === -1) {
+          return `'${args.options.scope}' is not a valid application customizer scope. Allowed values are: ${this.allowedScopes.join(',')}`;
+        }
+
+        if (!args.options.newTitle && !args.options.clientSideComponentProperties) {
+          return `Please specify option to be updated`;
+        }
+
+        return validation.isValidSharePointUrl(args.options.webUrl);
+      }
+    );
+  }
+
+  #initOptionSets(): void {
+    this.optionSets.push(
+      { options: ['id', 'title', 'clientSideComponentId'] }
+    );
+  }
+
+  public async commandAction(logger: Logger, args: CommandArgs): Promise<void> {
+    const { clientSideComponentId, title, id, webUrl }: Options = args.options;
+
+    if (this.verbose) {
+      logger.logToStderr(`Updating application customizer '${clientSideComponentId || title || id}' on the site '${webUrl}'...`);
+    }
+
+    try {
+      await this.updateAppCustomizer(args);
+    }
+    catch (err: any) {
+      this.handleRejectedODataJsonPromise(err);
+    }
+  }
+
+  private async updateAppCustomizer(args: CommandArgs): Promise<void> {
+    const requestBody: any = {};
+
+    if (args.options.newTitle) {
+      requestBody.Title = args.options.newTitle;
+    }
+
+    if (args.options.clientSideComponentProperties) {
+      requestBody.ClientSideComponentProperties = args.options.clientSideComponentProperties;
+    }
+
+    const appCustomizer = await this.getAppCustomizerToUpdate(args.options);
```

Comment:
Lets move this function call to the commandAction method so that each function does what it says.